cmd/otel-allocator/server/server.go

```diff
@@ -48,26 +47,19 @@ type collectorJSON struct {
 	Jobs []*target.Item `json:"targets"`
 }
 
-type DiscoveryManager interface {
-	GetScrapeConfigs() map[string]*promconfig.ScrapeConfig
-}
-
 type Server struct {
-	logger           logr.Logger
-	allocator        allocation.Allocator
-	discoveryManager DiscoveryManager
-	server           *http.Server
+	logger    logr.Logger
+	allocator allocation.Allocator
+	server    *http.Server
 
-	compareHash          uint64
 	scrapeConfigResponse []byte
```

Comment:
I think we need a lock around the scrape config response to prevent concurrent access from the discovery manager?